server/public/model/client4_test.go

```diff
@@ -149,3 +153,30 @@ func TestClient4RequestCancellation(t *testing.T) {
 		<-done
 	})
 }
+
+func ExampleClient4_GetUsers() {
+	client := model.NewAPIv4Client("http://localhost:8065")
+	client.SetToken(os.Getenv("MM_TOKEN"))
+
+	const perPage = 100
+	var page int
+	var allUsers []*model.User
+	for {
+		users, _, err := client.GetUsers(context.TODO(), page, perPage, "")
+		if err != nil {
+			log.Printf("error fetching users: %v", err)
+			return
+		}
+
+		if len(users) == 0 {
+			break
+		}
```

Comment:
I generally tend to break as soon as the length is less than the page size. That can prevent making an additional call and potentially getting an error (depending on how the call is implemented that is).

Reply:
Good call :+1: